sklearnex/tests/utils/base.py

```diff
@@ -344,3 +346,127 @@ def gen_dataset(
     np.uint32,
     np.uint64,
 ]
+
+
+def _get_processor_info():
+    proc = ""
+    if platform.system() == "Linux":
+        proc = (
+            subprocess.check_output(["/usr/bin/cat", "/proc/cpuinfo"])
+            .strip()
+            .decode("utf-8")
+        )
+    elif platform.system() == "Windows":
+        proc = platform.processor()
+    elif platform.system() == "Darwin":
+        proc = (
+            subprocess.check_output(["/usr/bin/sysctl", "-n", "machdep.cpu.brand_string"])
+            .strip()
+            .decode("utf-8")
+        )
+
+    return proc
+
+
+_NONINTEL_DESELECTIONS = [
```

Comment:
almost all tests for the certain algos disabled I would suggest use condition check for the estimator and disable it in the test suit instead of having this deselections here. Expecting less code for this